### PR TITLE
Added AsrSignal support.

### PR DESCRIPTION
**Pull request by DmytroDev**

### CODE REVIEW

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/rcml/domain/SupportedAsrLanguages.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Library class with all supported languages.
+ *
+ * @author Dmitriy Nadolenko
+ */
+public class SupportedAsrLanguages {
```
Comment (gvagenas) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/rcml/domain/SupportedAsrLanguages.java]: @DmytroDev I believe we should have the languages at the restcomm.xml and load them with configuration, similar to drivers. This way we add add/remove or enable/disable languages without changing source code. Wdyt?

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java [hunk 2]
```diff
@@ -1517,11 +1531,42 @@ public Gathering(final ActorRef source) {
 
         @Override
         public void execute(final Object message) throws Exception {
+
+            // check mandatory attribute partialResultCallback
+            Attribute resultCallbackAttr = verb.attribute(GatherAttributes.ATTRIBUTE_PARTIAL_RESULT_CALLBACK);
             final NotificationsDao notifications = storage.getNotificationsDao();
+            if (resultCallbackAttr == null) {
```
Comment (gvagenas) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java]: @DmytroDev why when `ATTRIBUTE_PARTIAL_RESULT_CALLBACK` is null we stop Interpreter? I believe we should stop only when `ATTRIBUTE_ACTION` is null. There might be cases where application doesn't need to know about partial results but for the final result. Isn't that the case?

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java [hunk 3]
```diff
@@ -1517,11 +1531,42 @@ public Gathering(final ActorRef source) {
...
+
+            // parse attribute "language"
+            Attribute langAttr = verb.attribute(GatherAttributes.ATTRIBUTE_LANGUAGE);
+            String defaultLang = SupportedAsrLanguages.DEFAULT_LANGUAGE;
+            String lang = parseAttrLanguage(langAttr, defaultLang);
+
+            // parse attribute "hints"
+            String hints = null;
+            Attribute hintsAttr = verb.attribute(GatherAttributes.ATTRIBUTE_HINTS);
+            if (hintsAttr != null && !hintsAttr.value().isEmpty()) {
```
Comment (gvagenas) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java]: @DmytroDev shouldn't we stop interpreter when `hints==null`?

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java [hunk 4]
```diff
@@ -775,35 +783,44 @@ private void onMediaGroupResponse(Object message) throws TransitionFailedExcepti
...
-                    collectedDigits.append(dtmfResponse.get());
-                    // Collected digits == requested num of digits the complete the collect digits
-                    if (numberOfDigits != Short.MAX_VALUE) {
-                        if (collectedDigits.length() == numberOfDigits) {
-                            dtmfReceived = true;
-                            fsm.transition(message, finishGathering);
+            else if (is(gathering) || is(continuousGathering) || is(finishGathering)) {
+                final MediaGroupResponse<CollectedResult> dtmfResponse = (MediaGroupResponse<CollectedResult>) message;
+                CollectedResult data = dtmfResponse.get();
+                logger.info("!!!" + VoiceInterpreter.class.getName() + ": CollectedResult = " + data);
```
Comment (gvagenas) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java]: @DmytroDev please surround logger.info and logger.debug with `if (logger.isInfoEnabled)`

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java (hunk 3 above)
Comment (hamsterksu) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java]: @gvagenas  i think it's optional parameter. @YevgenL FYI

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/rcml/domain/SupportedAsrLanguages.java (hunk 1 above)
Comment (hamsterksu) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/rcml/domain/SupportedAsrLanguages.java]: @gvagenas we used existing approach. but xml config is more flexible.

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java (hunk 2 above)
Comment (hamsterksu) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java]: @gvagenas @YevgenL  we should stop it for `SPEECH` and `DTMF_SPEECH`
so let's add statement `inputType != DTMF` too

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java (hunk 4 above)
Comment (hamsterksu) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java]: @YevgenL let's remove this output

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java (hunk 2 above)
Comment (YevgenL) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java]: Changed to

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java (hunk 4 above)
Comment (YevgenL) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java]: Removed

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java (hunk 3 above)
Comment (YevgenL) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java]: 'hints' is optional parameter

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/rcml/domain/SupportedAsrLanguages.java (hunk 1 above)
Comment (YevgenL) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/rcml/domain/SupportedAsrLanguages.java]: Refactored ASR Languages as xml config

**Comment by hamsterksu** [comment]
@gvagenas please cancel/close this PR.
we have created new one - https://github.com/RestComm/Restcomm-Connect/pull/2329

**Pull request closed without merging**